crates/uv-pep508/src/marker/algebra.rs

```diff
@@ -421,6 +421,65 @@ impl InternerGuard<'_> {
         self.create_node(node.var.clone(), children)
     }
 
+    /// Returns a new tree where the only nodes remaining are non-`extra`
+    /// nodes.
+    ///
+    /// If there are only `extra` nodes, then this returns a tree that is
+    /// always true.
+    ///
+    /// This works by assuming all `extra` nodes are always true.
+    pub(crate) fn without_extras(&mut self, mut i: NodeId) -> NodeId {
+        if matches!(i, NodeId::TRUE | NodeId::FALSE) {
+            return i;
+        }
+
+        let parent = i;
+        let node = self.shared.node(i);
+        if matches!(node.var, Variable::Extra(_)) {
+            i = NodeId::FALSE;
+            for child in node.children.nodes() {
+                i = self.or(i, child.negate(parent));
+            }
```

Comment:
I'm trying to grok what this function does. What would it do with `((os_name == ... and extra == foo) or (sys_platform == ... and extra != foo))`?

Reply:
It would give you `os_name == ... or sys_platform == ...` back. Basically, it removes all `extra` nodes by assuming they are `true`.

Reply:
I added this as an example to the docs of this function.